core/src/main/java/com/netease/arctic/utils/StatisticsFileUtil.java

```diff
@@ -0,0 +1,287 @@
+        long fileSize = puffinWriter.fileSize();
+        long footerSize = puffinWriter.footerSize();
+        List<org.apache.iceberg.BlobMetadata> collect =
+            blobMetadata.stream().map(GenericBlobMetadata::from).collect(Collectors.toList());
+        return new GenericStatisticsFile(
+            snapshotId, outputFile.location(), fileSize, footerSize, collect);
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      } finally {
+        close();
+      }
+    }
+
+    private void checkNotClosed() {
+      Preconditions.checkState(!closed, "Cannot operate on a closed writer");
+    }
+
+    private void close() {
+      if (!closed) {
+        closed = true;
+        try {
+          puffinWriter.close();
+        } catch (IOException e) {
+          throw new UncheckedIOException(e);
+        }
+      }
+    }
+  }
+
+  public static class Reader {
+    private final Table table;
+
+    private Reader(Table table) {
+      this.table = table;
+    }
+
+    public <T> List<T> read(
+        StatisticsFile statisticsFile, String type, DataSerializer<T> deserializer) {
+      return read(statisticsFile, type).stream()
+          .map(deserializer::deserialize)
+          .collect(Collectors.toList());
+    }
+
+    public List<ByteBuffer> read(StatisticsFile statisticsFile, String type) {
+      try (PuffinReader puffin =
+          Puffin.read(table.io().newInputFile(statisticsFile.path())).build()) {
+        FileMetadata fileMetadata = puffin.fileMetadata();
+        List<BlobMetadata> blobs =
+            fileMetadata.blobs().stream()
+                .filter(b -> type.equals(b.type()))
+                .collect(Collectors.toList());
+        return Lists.newArrayList(Iterables.transform(puffin.readAll(blobs), Pair::second));
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    }
+  }
+
+  /**
+   * Copy a statistic file with a new snapshot id, and it points to the same file in the file system
+   * as the original file.
+   *
+   * @param statisticsFile - original statistic file
+   * @param snapshotId - new snapshot id
+   * @return a new copied statistic file
+   */
+  public static StatisticsFile copyToSnapshot(StatisticsFile statisticsFile, long snapshotId) {
+    return new GenericStatisticsFile(
+        snapshotId,
+        statisticsFile.path(),
+        statisticsFile.fileSizeInBytes(),
+        statisticsFile.fileFooterSizeInBytes(),
+        statisticsFile.blobMetadata());
+  }
+
+  /**
+   * Find the statistics file that satisfies the condition and belongs to the latest snapshot.
+   *
+   * @param table - Iceberg Table
+   * @param currentSnapshotId - find from this snapshot to its ancestors
+   * @param condition - the condition to filter the statistics file
+   * @return the statistics file, return null if not found
+   */
+  public static List<StatisticsFile> findLatestValidStatisticsFiles(
```

Comment:
Most of the time we need to find the static file belonging to a certain snapshot to ensure the correctness of the logic. Searching for its parent's static file only makes sense in some special scenarios when it cannot be found.

Reply:
I introduce a mark in snapshot summary in order to find the snapshot.